src/js/components/editor/Editor.vue

```diff
@@ -0,0 +1,284 @@
+<template>
+    <div class="text-pane" :class='{fullscreen}'>
+        <toolbar :state="state"
+            @fullscreen="toggleFullScreen"
+        />
+        <div v-if="!colInRouter">
+            <text-selector @selectedColumn="getText"></text-selector>
+        </div>
+        <section class="editor">
+            <div class="editor-column" v-for="(column, columnIndex) of currentText.cols" :data-col="column.col" :key="columnIndex + column.col">
+                <input id="editor-input" class="" @keyup="onKeyup" @blur="onInputBlur" v-model="input" />
+                <table class="editor-table" >
+                    <thead>
+                        <th>{{column.col}}</th>
+                        <th class="line-col"></th>
+                    </thead>
+                    <tbody>
+                        <tr v-for="(line, lineIDX) of column.lines" :key="lineIDX + line.lineId" :data-line="line.line" :data-line-id="line.lineId"
+                            dir="rtl">
+                            <td class="text-col">
+                                <sqe-sign v-for="sign of line.signs"
+                                    @click="onClick"
+                                    :sign="sign"
+                                    :state="state" 
+                                    :focusedSignId="focusedSignId"
+                                    :key="sign.id"
+                                />
+                            </td>
+                            <td class="line-col">{{line.line}}</td>
+                        </tr>
+                    </tbody>
+                </table>
+            </div>
+        </section>
+    </div>
+</template>
+
+<script>
+
+import KEY_CODES from './key_codes.js';
+import TextSelector from './TextSelector.vue'
+import Toolbar from './Toolbar.vue'
```

Comment:
Maybe we change the name to something like TextEditorToolbar?  Maybe we should standardize so that the toolbar for each pane has the same sort of name.  What do you think?

Reply:
I'm going to add a card to handle the toolbar for system-wide consistency ... I'm actually wondering if we might want to introduce some sort of base component to create a common toolbar. In any case, let's chew on that.

Reply:
Great idea!